odl/contrib/datasets/ct/mayo.py

```diff
@@ -0,0 +1,270 @@
+# Copyright 2014-2017 The ODL contributors
+#
+# This file is part of ODL.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+"""Tomographic datasets from Mayo Clinic.
+
+In addition to the standard ODL requirements, this library also requires:
+
+    - tqdm
+    - dicom
+    - A copy of the Mayo dataset, see
+    https://www.aapm.org/GrandChallenge/LowDoseCT/#registration
+"""
+
+from __future__ import division
+import numpy as np
+import os
+import dicom
+import odl
+import tqdm
+
+from dicom.datadict import DicomDictionary, NameDict, CleanName
+from odl.contrib.datasets.ct.mayo_dicom_dict import new_dict_items
+
+# Update the dicom dictionary with the extra Mayo tags
+DicomDictionary.update(new_dict_items)
+new_names_dict = dict([(CleanName(tag), tag) for tag in
+                       new_dict_items])
+NameDict.update(new_names_dict)
+
+
+__all__ = ('projections_from_folder', 'volume_from_folder')
+
+
+def _read_projections(folder, proj_start=1, proj_end=-1):
+    """Read mayo projections from a folder."""
+    projections = []
+    datasets = []
+
+    # Get the relevant file names
+    file_names = sorted([f for f in os.listdir(folder) if f.endswith(".dcm")])
+
+    if len(file_names) == 0:
+        raise ValueError('No dicom files found in {}'.format(folder))
+
+    file_names = file_names[proj_start:proj_end]
+
+    for file_name in tqdm.tqdm(file_names, 'loading projection data'):
+        # read the file
+        dataset = dicom.read_file(folder + '/' + file_name)
+
+        # Get some required data
+        rows = dataset.NumberofDetectorRows
+        cols = dataset.NumberofDetectorColumns
+        hu_factor = dataset.HUCalibrationFactor
+        rescale_intercept = dataset.RescaleIntercept
+        rescale_slope = dataset.RescaleSlope
+
+        # print(rescale_intercept, rescale_slope)
+
+        # Load the array as bytes
+        data_array = np.array(np.frombuffer(dataset.PixelData, 'H'),
+                              dtype='float32')
+        data_array = data_array.reshape([rows, cols], order='F').T
+
+        # Rescale array
+        # rescaled_array = rescale_intercept + data_array * rescale_slope
```

Comment:
`hu_factor` missing. But I think the code is pretty obvious, the comment is not really needed